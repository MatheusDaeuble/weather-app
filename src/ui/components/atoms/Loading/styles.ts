import styled from 'styled-components/native';

export const Container = styled.View<{ flex1: boolean }>`
  ${({ flex1 }) => (flex1 ? 'flex: 1;' : '')}
  justify-content: center;
  align-items: center;
`;
