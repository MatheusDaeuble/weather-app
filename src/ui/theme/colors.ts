export enum Colors {
  // PRIMARY
  PRIMARY = '#2E89DF',

  // SECONDARY
  SECONDARY = '#809FFF',

  WHITE = '#FFFFFF',
  BLACK = '#000000',

  // FEEDBACK COLORS
  SUCCESS = '#0E934C',
  WARNING = '#FF8B2C',
  DANGER = '#EB5757',

  // GRAY SCALE
  GRAY_DARK = '#4B4F58',
  GRAY = '#949494',
  GRAY_LIGHT = '#CECECE',
}

export type Color =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black'
  | 'success'
  | 'warning'
  | 'danger'
  | 'gray-dark'
  | 'gray'
  | 'gray-light';

export const colors: { [key in Color]: string } = {
  'primary': Colors.PRIMARY,
  'secondary': Colors.SECONDARY,
  'white': Colors.WHITE,
  'black': Colors.BLACK,
  'success': Colors.SUCCESS,
  'danger': Colors.DANGER,
  'warning': Colors.WARNING,
  'gray-dark': Colors.GRAY_DARK,
  'gray': Colors.GRAY,
  'gray-light': Colors.GRAY_LIGHT,
};

export const defaultTheme = {
  colors,
};
