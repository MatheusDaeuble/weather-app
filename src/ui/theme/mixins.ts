import { Dimensions, PixelRatio } from 'react-native';

const { width: WINDOW_WIDTH, height: WINDOW_HEIGHT } = Dimensions.get('window');
const guidelineBaseWidth = 375;
const guidelineBaseHeight = 700;

export const scaleSize = (size: number) =>
  Number(((WINDOW_WIDTH / guidelineBaseWidth) * size).toFixed(0));

export const scaleHeight = (size: number) =>
  Number(((WINDOW_HEIGHT / guidelineBaseHeight) * size).toFixed(0));

export const scaleFont = (size: number) =>
  Number((size * PixelRatio.getFontScale()).toFixed(0));
