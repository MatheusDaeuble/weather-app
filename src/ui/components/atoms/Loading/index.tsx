import React from 'react';
import { ActivityIndicator, ActivityIndicatorProps } from 'react-native';
import { Color, colors } from 'ui/theme/colors';
import { Container } from './styles';

export interface LoadingProps extends ActivityIndicatorProps {
  color?: Color;
  flex1?: boolean;
}

const Loading = ({
  color = 'primary',
  size = 'large',
  flex1 = true,
  style,
  ...props
}: LoadingProps) => {
  return (
    <Container style={style} flex1={flex1}>
      <ActivityIndicator color={colors[color]} size={size} {...props} />
    </Container>
  );
};

export default Loading;
