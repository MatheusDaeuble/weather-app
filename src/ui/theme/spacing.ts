import { scaleSize } from './mixins';

export const SCALE_28 = scaleSize(28);
export const SCALE_24 = scaleSize(24);
export const SCALE_20 = scaleSize(20);
export const SCALE_16 = scaleSize(16);
export const SCALE_12 = scaleSize(12);
export const SCALE_8 = scaleSize(8);
export const SCALE_4 = scaleSize(4);
