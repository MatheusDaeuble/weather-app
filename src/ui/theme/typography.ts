import { scaleFont } from './mixins';

// FONT FAMILY
export const FONT_FAMILY_REGULAR = 'OpenSans_400Regular';
export const FONT_FAMILY_BOLD = 'OpenSans_700Bold';

// FONT SIZE
export const FONT_SIZE_64 = scaleFont(62);
export const FONT_SIZE_26 = scaleFont(26);
export const FONT_SIZE_18 = scaleFont(18);
export const FONT_SIZE_16 = scaleFont(16);
export const FONT_SIZE_14 = scaleFont(14);

// LINE HEIGHT
export const LINE_HEIGHT_93 = scaleFont(93);
export const LINE_HEIGHT_39 = scaleFont(39);
export const LINE_HEIGHT_27 = scaleFont(27);
export const LINE_HEIGHT_24 = scaleFont(24);
export const LINE_HEIGHT_21 = scaleFont(21);
