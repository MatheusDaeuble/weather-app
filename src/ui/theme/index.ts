import * as Colors from './colors';
import * as Spacing from './spacing';
import * as TypographyAttributes from './typography';
import * as Mixins from './mixins';

export { TypographyAttributes, Spacing, Colors, Mixins };
