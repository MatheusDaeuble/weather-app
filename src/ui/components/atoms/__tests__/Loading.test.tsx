import React from 'react';
import renderer from 'react-test-renderer';
import { Color, colors } from 'ui/theme/colors';
import Loading, { LoadingProps } from '../Loading';

describe('Loading Atom', () => {
  describe('Render color', () => {
    Object.keys(colors).forEach((color: Color) => {
      it(`Render color: ${color}`, () => {
        const component = renderer.create(<Loading color={color} />).toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('Render size', () => {
    ['small', 'large'].forEach(size => {
      it(`Render size: ${size}`, () => {
        const component = renderer
          .create(<Loading size={size as LoadingProps['size']} />)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });
});
