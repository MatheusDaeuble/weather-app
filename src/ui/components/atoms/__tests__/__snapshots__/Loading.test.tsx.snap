// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Atom Render color Render color: black 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#000000"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: danger 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#EB5757"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: gray 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#949494"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: gray-dark 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#4B4F58"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: gray-light 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#CECECE"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: primary 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#2E89DF"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: secondary 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#809FFF"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: success 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#0E934C"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: warning 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#FF8B2C"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render color Render color: white 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#FFFFFF"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render size Render size: large 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#2E89DF"
    size="large"
  />
</View>
`;

exports[`Loading Atom Render size Render size: small 1`] = `
<View
  flex1={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="#2E89DF"
    size="small"
  />
</View>
`;
